File: challenges/notification-dispatcher/python/starter/exercise.py
"""Implement the contract in the challenge README. Reference code is separate."""
from tooling.python.contracts import JobQueuePort, KeyValueStore, Sender

class NotificationDispatcher:

    def __init__(self, store: KeyValueStore, queue: JobQueuePort, sender: Sender, idempotency_ttl_ms: int):
        raise NotImplementedError('Not implemented: __init__')

    def submit(self, key: str, message: str) -> str:
        raise NotImplementedError('Not implemented: submit')

    def process_one(self) -> str:
        raise NotImplementedError('Not implemented: process_one')

File: challenges/notification-dispatcher/typescript/starter/exercise.ts
import type { JobQueuePort, KeyValueStore, Sender } from "../../../../tooling/typescript/contracts.js";

// Implement the challenge contract here.
export class NotificationDispatcher {
  constructor(store: KeyValueStore, queue: JobQueuePort,
              sender: Sender, idempotencyTtlMs: number) {
    throw new Error("Not implemented: constructor");
  }

  submit(key: string, message: string): string {
    throw new Error("Not implemented: submit");
  }

  processOne(): string {
    throw new Error("Not implemented: processOne");
  }
}
